add glossary and faqs (#786)
LF-Decentralized-Trust-labs/paladin: doc-site/docs/faqs.md

# Frequently Asked Questions

### What is Paladin?

**Paladin is a framework for building private, programmable applications on top of standard EVM blockchains.** It provides a full stack of technologies to manage the complex, off-chain execution of private transactions while using a public or private blockchain for final settlement, ensuring that confidential data remains private without sacrificing interoperability.

### How does Paladin compare to other privacy solutions?

**Paladin's unique value is its role as an integrator and a standards-setter, not a single, monolithic privacy technology.**

* Unlike privacy coins (e.g., Zcash) which focus only on anonymous payments, Paladin is a full-stack framework for building complex, private applications.
* Unlike previous private EVM implementations, Paladin runs on top of **un-modified** EVM chains and offers a more scalable and robust model for private transactions.
* Instead of being a single solution, Paladin is designed to host many different privacy technologies (`noto`, `zeto`, etc.) in a single runtime, with the goal of creating common standards for private assets, similar to what ERC-20 did for public tokens.

> ### What is the difference between Paladin and a Layer 2 (L2)?

Reviewer: ```suggestion
### What is the relationship between Paladin, permissioned chains, L1s and L2s?
```

**Paladin is a client for enabling programmable privacy, whereas L2s are primarily designed to provide scalability.**

> * **Layer 2s** exist to "roll up" multiple transactions to solve for network congestion and high fees. They are not inherently privacy technologies.

Reviewer: ```suggestion
```

> * **Paladin** is focused on confidentiality. Every private transaction it facilitates is backed 1-to-1 by a transaction on the underlying base ledger.

Reviewer: ```suggestion
```

> ### Paladin is additive to L2s; it can run on top of an L2 network just as easily as it can on a Layer 1.

Reviewer: ```suggestion
```

> Why does Paladin use a base public ledger instead of being its own standalone blockchain?

Reviewer: ```suggestion
Why does Paladin use a base ledger instead of being its own standalone blockchain?
```

* **Standardization:** It avoids creating another siloed network and instead focuses on providing a common privacy standard for the entire, vast EVM ecosystem.

### Where is the private data actually stored?

**Private transaction data is never written to the public blockchain; it is exchanged directly between participants and stored locally.** The private details are sent securely between the Paladin runtimes involved in a transaction and kept in each participant's local **State Store**. Only a cryptographic proof (a hash) is sent to the base ledger. This proof acts as a public witness, preventing double-spends and ensuring transactional integrity without revealing any of the confidential data.

### What is a Domain and what is its role in the architecture?

**A Domain is an isolated, pluggable execution engine that implements a specific set of privacy-preserving features.** Like a secured workspace within a larger office, a Domain provides a specialized environment tailored for a specific task. This model allows developers to choose the right privacy tool for the job—whether it's for tokens or general smart contracts—without being locked into a single technology.

> ### What is the difference between the `noto`, `pente`, and `zeto` domains?

Reviewer: ```suggestion
### What is the difference between the `noto`, `pente`, and `zeto` reference domains?
```

**These domains provide different tools and trust models for different privacy use cases.**

* **`noto`**: Uses a **Trusted Notary** to manage confidential assets. This is ideal for ecosystems where a central rule-keeper is required to enforce specific rules.
* **`pente`**: Provides **Private EVM Execution** for running complex, confidential business logic and workflows, such as a Delivery versus Payment (DvP) agreement.
* **`zeto`**: Leverages **Zero-Knowledge Proofs** for maximum asset privacy, enabling transactions where no underlying data is revealed to any party.

> 

Reviewer: ```suggestion

These domains are _reference_ domains and have been implemented using Paladin's pluggable architecture. Other domains can be implemented to the same plugin interface, for example a zero-knowledge token that uses a different ZK standard to the one used by **zeto**.
```

### Where is the best place to ask questions or get help?

**The best place to connect with the Paladin community and development team is on the Hyperledger Discord server.** This is the primary forum for real-time discussions, questions, and sharing ideas.

* Join the conversation here: **[Hyperledger Discord](https://discord.gg/hyperledger)** (look for the Paladin channel).

### How should I report a bug?

**Bugs should be reported by creating a new issue in the official GitHub repository.** When reporting a bug, please provide as much detail as possible, including your operating system, project version, and clear steps to reproduce the issue.

* Report a bug here: **[Paladin GitHub Issues](https://github.com/LF-Decentralized-Trust-labs/paladin/issues)**

### What are the guidelines for contributing to Paladin?

**Contributions from the community are highly encouraged and welcome.** The general process involves forking the repository, creating a feature branch for your changes, and submitting a pull request for review. All code contributions must include tests and a "Signed-off-by" line in the commit message to comply with the Developer Certificate of Origin (DCO).

> * Explore the code and contribute here: **[Paladin GitHub Repository](https://github.com/LF-Decentralized-Trust-labs/paladin)**

Reviewer: It might be good to add a contributing file such as https://github.com/hyperledger/firefly/blob/main/CONTRIBUTING.md and then reference it here.

In fact more specifically we should probably review everything in https://github.com/LF-Decentralized-Trust/governance/tree/main/tac/guidelines and start aligning the project to those guidelines. Happy for this PR to merge without that but wanted to mention.
